circuits/cpp/src/aztec3/circuits/rollup/base/native_base_rollup_circuit.cpp

```diff
@@ -77,20 +77,11 @@ std::vector<NT::fr> calculate_contract_leaves(BaseRollupInputs const& baseRollup
 
         // loop over the new contracts
         // TODO: NOTE: we are currently assuming that there is only going to be one
-        for (auto& new_contact : new_contacts) {
-            NT::address contract_address = new_contact.contract_address;
-            NT::address portal_contract_address = new_contact.portal_contract_address;
-            NT::fr const function_tree_root = new_contact.function_tree_root;
-
-            // Pedersen hash of the 3 fields (contract_address, portal_contract_address, function_tree_root)
-            auto contract_leaf = crypto::pedersen_commitment::compress_native(
-                { contract_address, portal_contract_address, function_tree_root }, GeneratorIndex::CONTRACT_LEAF);
-
+        for (auto& leaf_preimage : new_contacts) {
             // When there is no contract deployment, we should insert a zero leaf into the tree and ignore the
             // member-ship check. This is to ensure that we don't hit "already deployed" errors when we are not
             // deploying contracts. e.g., when we are only calling functions on existing contracts.
-            auto to_push = contract_address == NT::address(0) ? NT::fr(0) : contract_leaf;
-
+            auto to_push = leaf_preimage.contract_address == NT::address(0) ? NT::fr(0) : leaf_preimage.hash();
```

Comment:
Maybe add an is_empty method to the class, like we're doing for other structs? This way we don't have different criteria for emptiness spread throughout the code.

Reply:
Fine idea. It is currently used by cbinds and probably also other places. So will create a separate issue for it instead, see #459.